File: backend/src/domain/entities/LlmProviderConfig.ts
export type LlmProviderConfig = {
  id: string;
  key: string;
  provider: string;
  displayName: string;
  baseUrl: string | null;
  apiKeyEnvVar: string | null;
  model: string;
  inputCostPer1K: number;
  outputCostPer1K: number;
  enabled: boolean;
  isDefault: boolean;
  notes: string | null;
  createdAt: Date;
  updatedAt: Date;
};

File: app/models/llm_provider_config.rb
class LlmProviderConfig < ApplicationRecord
  validates :key, presence: true, uniqueness: true
  validates :provider, presence: true
  validates :display_name, presence: true
  validates :model, presence: true

  def self.default_enabled
    where(enabled: true, is_default: true).order(created_at: :asc).first
  end
end
